src/python/pants/backend/python/subsystems/python_native_code.py

```diff
@@ -64,7 +64,7 @@ def native_target_has_native_sources(self, target):
   def _native_target_matchers(self):
     return {
       Exactly(PythonDistribution): self.pydist_has_native_sources,
```

Comment:
In the current world, all public targets exposed by register.py's can be subclasses for purposes you know not making Exactly on these Target types probably never advised. It seems like you'd stll want to operate on a `class FoursquarePythonDistribution(PythonDistribution)` that might add information used by a Foursquare custom task or just act like a macro, sealing in certain Foursquare specific values.

Reply:
You're totally right. I will make that change.

Reply:
Done in ef41174, along with making `_PYTHON_PLATFORM_TARGETS_CONSTRAINT = SubclassesOf(PythonBinary, PythonDistribution)` (instead of `Exactly` as well). I assumed that change would be correct for the same reasons.